### PR TITLE
refact: improve null analysis

### PR DESCRIPTION
**Pull request by sebthom**
This PR bumps no-npe to version 1.1.0 (which contains more refined external null annotation) and applies all necessary code changes to satisfy null analysis checks by the Eclipse compiler.

### CODE REVIEW

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java
Comment (rubenporras) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java:586]: I have already realized this `return null`, and I agree that this seems better, but it is a breaking change for any consumer that was handling the null case.

Personally I am okay with it, I know we have code that relies on null, but I also know how to refactor. However, I it might break other projects.

Therefore I would argue we should let it be. @mickaelistria , any opinions?

File: org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/debugmodel/DSPBreakpointManager.java
Comment (rubenporras) [inline org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/debugmodel/DSPBreakpointManager.java:71]: What is the reason for it? A limitation in the JDT?

Is Void not type used to express that the method does not return a value, that is more or less equivalent to null?

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java
Comment (rubenporras) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java:282]: this changes to return null seem to make the code more inline with the documentation of the method we are overriding, and I think it would be nice to have a separate PR for them that we can merge independently.

File: org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/launcher/DSPLaunchDelegate.java
Comment (rubenporras) [inline org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/launcher/DSPLaunchDelegate.java:194]: can we have another PR for this one? It looks independent to me as well

File: org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/debugmodel/DSPBreakpointManager.java
Comment (sebthom) [reply org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/debugmodel/DSPBreakpointManager.java:71]: Void futures are completed with a null value. Otherwise you cannot complete it.

File: org.eclipse.lsp4e.jdt/src/org/eclipse/lsp4e/jdt/LSJavaCompletionProposalComputer.java
Comment (rubenporras) [inline org.eclipse.lsp4e.jdt/src/org/eclipse/lsp4e/jdt/LSJavaCompletionProposalComputer.java:108]: I think we can also merge this fix already if you like to have another PR.

**Comment by rubenporras** [comment]
@sebthom , if you like to do a new PR with the three changes which are clear and not just flipping annotations, I think that would be good.

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java
Comment (sebthom) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java:586]: I have no hard feelings about this change. I can revert it if required.

File: org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/launcher/DSPLaunchDelegate.java
Comment (sebthom) [reply org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/launcher/DSPLaunchDelegate.java:194]: no, List.of() does not compile if it is passed a nullable value.

**Review by rubenporras — APPROVED** [review]

**Comment by sebthom** [comment]
@rubenporras we need a new approval from IP team, see https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/16066

I hope that happens soon.

**Comment by rubenporras** [comment]
> > @sebthom : 6,615 [ERROR] Failed to execute goal org.eclipse.dash:license-tool-plugin:1.1.1-SNAPSHOT:license-check (default-cli) on project parent: Some dependencies must be vetted. -> [Help 1]
> 
> Yes I already opened a ticket. see my previous ocmment.

Sorry, I cross-posted.

**Pull request merged**
